### PR TITLE
feat(server): game hub TS types generator

### PR DESCRIPTION
**Pull request by Alequez97**

### CODE REVIEW

File: GomokuClient/packages/gomoku-core/src/types/hubs/GomokuServer.Api.Hubs.Messages.Client.ts
Comment (vkuprin) [inline GomokuClient/packages/gomoku-core/src/types/hubs/GomokuServer.Api.Hubs.Messages.Client.ts:1]: I would prefer all generated type to be in src/api/generated

File: package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "private": true,
   "scripts": {
     "run:server": "dotnet run --cofiguration Release --project ./GomokuServer/src/GomokuServer.Api/GomokuServer.Api.csproj",
+    "generate-hub-types:server": "dotnet tsrts --project ./GomokuServer/src/GomokuServer.Api/GomokuServer.Api.csproj --output ./GomokuClient/packages/gomoku-core/src/types/hubs  --assemblies",
```
Comment (vkuprin) [inline package.json:5]: Please also extend Makefile to add this script

File: GomokuClient/packages/gomoku-core/src/types/hubs/TypedSignalR.Client/index.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      receiver.sendMessage(...args);
+    const __gameHubError = (...args: [ErrorMessage]) =>
+      receiver.gameHubError(...args);
+
+    connection.on("GameGroupJoined", __gameGroupJoined);
+    connection.on("GameStarted", __gameStarted);
+    connection.on("PlayerMadeMove", __playerMadeMove);
+    connection.on("SendMessage", __sendMessage);
+    connection.on("GameHubError", __gameHubError);
+
```
Comment (vkuprin) [inline GomokuClient/packages/gomoku-core/src/types/hubs/TypedSignalR.Client/index.ts:133]: Nice 👌

File: package.json (hunk 1 above)
Comment (Alequez97) [reply package.json]: done

File: GomokuClient/packages/gomoku-core/src/types/hubs/GomokuServer.Api.Hubs.Messages.Client.ts
Comment (Alequez97) [reply GomokuClient/packages/gomoku-core/src/types/hubs/GomokuServer.Api.Hubs.Messages.Client.ts:1]: done

File: GomokuClient/packages/gomoku-core/src/types/hubs/TypedSignalR.Client/index.ts (hunk 2 above)
Comment (Alequez97) [reply GomokuClient/packages/gomoku-core/src/types/hubs/TypedSignalR.Client/index.ts]: Yeah, looks good

**Review by vkuprin — APPROVED** [review]

**Pull request merged**